Tests/MapboxCoreNavigationTests/NavigationServiceTests.swift

```diff
@@ -503,31 +503,45 @@ class NavigationServiceTests: XCTestCase {
         let now = Date()
         let trace = Fixture.generateTrace(for: route).shiftedToPresent()
 
-        trace.forEach { navigation.router.locationManager!(navigation.locationManager, didUpdateLocations: [$0]) }
+        trace.forEach {
+            navigation.router.locationManager!(navigation.locationManager, didUpdateLocations: [$0])
+            RunLoop.main.run(until: Date().addingTimeInterval(0.01))
```

Comment:
You can use function `waitForNavNativeCallbacks(timeout: 0.01)` from this file, or just remove this function and change all occurrences to `RunLoop.main.run`, if it will work.

Reply:
waitForNavNativeCallbacks doesn't wait for callback though? It is just the same runloop. We can back to this once all tests are working.